File: backend-api/app/Models/Member.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Member extends Model
{
    use HasFactory, SoftDeletes;

    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'members';

    /**
     * The primary key for the model.
     *
     * @var string
     */
    protected $primaryKey = 'id';

    /**
     * Indicates if the IDs are auto-incrementing.
     *
     * @var bool
     */
    public $incrementing = false;

    /**
     * The data type of the auto-incrementing ID.
     *
     * @var string
     */
    protected $keyType = 'string';

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'name',
        'email',
        'mobile',
        'unit_id',
        'company_id',
        'unit_number',
        'is_active',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'is_active' => 'boolean',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    /**
     * Get the unit that owns the member.
     */
    public function unit()
    {
        return $this->belongsTo(Unit::class);
    }

    /**
     * Get the staff assigned to this member.
     */
    public function staff()
    {
        return $this->belongsToMany(Staff::class, 'member_staff_assignments')
            ->withTimestamps();
    }

    /**
     * Get the staff verified by this member.
     */
    public function verifiedStaff()
    {
        return $this->hasMany(Staff::class, 'verified_by_member_id');
    }

    /**
     * Get the staff created by this member.
     */
    public function createdStaff()
    {
        return $this->hasMany(Staff::class, 'created_by');
    }

    /**
     * Scope a query to only include active members.
     *
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }
}

File: lib/src/core/models/member.dart
/// Model representing a member
class Member {
  final String id;
  final String name;
  final String email;
  final String mobile;
  final String unitId;
  final String companyId;
  final String unitNumber;
  final bool isActive;
  
  Member({
    required this.id,
    required this.name,
    required this.email,
    required this.mobile,
    required this.unitId,
    required this.companyId,
    required this.unitNumber,
    required this.isActive,
  });
  
  factory Member.fromJson(Map<String, dynamic> json) {
    return Member(
      id: json['id'],
      name: json['name'],
      email: json['email'] ?? '',
      mobile: json['mobile'],
      unitId: json['unit_id'],
      companyId: json['company_id'],
      unitNumber: json['unit_number'],
      isActive: json['is_active'] ?? true,
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'mobile': mobile,
      'unit_id': unitId,
      'company_id': companyId,
      'unit_number': unitNumber,
      'is_active': isActive,
    };
  }
}
